### ISSUE TITLE
Nodes missing in kubernetes but az aks shows the nodes!

### ISSUE THREAD

**Issue by mdoulaty** [Azure/AKS#152]
Nodes in my kubernetes cluster are gone missing since yesterday. `kubernetes get nodes` shows 
`No resources found.`, but when I use `az aks list`, it shows 1 for node count:
```
    "agentPoolProfiles": [
      {
        "count": 1,
```
When I scale up to i.e. 2 nodes, az aks shows node count as 2, but kubectl only shows 1 node.
When I scale down to 1, `az aks` shows one node, but again kubernetes reports no resources found. That one node is somehow gone missing...
Seems like `az aks` is somehow not synced with kubernetes. Any ideas how to solve this issue?
A bit of background information:  the cluster is located in westeurope, kubernetes version was 1.8.2, it was setup around two weeks ago and everything was working okay until yesterday night and then the node is gone missing...

**Comment by vlesierse** [comment]
Same for me...

**Comment by OJezu** [comment]
We had three nodes configured. Portal reported that AKS is set to 3 agents, all three VM were "Available", had disks mounted etc. Service Health did not report any problems, Activity Log was empty.

Nodes disappeared from master configuration, along with all of the custom configuration. `kubectl get nodes` `kubectl get pods` `kubectl get deployments` `kubectl get services` all returned empty list. It seemed as if cluster was reset to default configuration, but without nodes connected. After adding new nodes, default services e.g. k8s dashboard came up online.

I've rescaled the cluster down to 1 node using `az aks scale`, and then I rescaled it up to 4 nodes . After that finished `kubectl get nodes` reported three new nodes. Last VM not connected to k8s master I've deleted in the Azure Portal.

When I started investigating the problem, I've re-applied our secrets, services, and deployment configuration, and only after pods were waiting for being provisioned for a few minutes, I've discovered no nodes were connected to the k8s master.

I had made one change to the resources provided by AKS: I had added a Service Endpoint to the vnet connecting all VMs, to make them able to connect to a firewalled Storage Account. That change was missing after the incident, and I've had to recreate it.

k8s version is 1.8.2
Region is West Europe
Seems like the cluster stopped activity at 22:50 27 January UTC

**Comment by vlesierse** [comment]
@OJezu our cluster died around the same time. Had it up & running by scaling down, back up and upgrade to 1.8.6. Then all the node came back. However the configuration was gone and needed to provision back everything.

**Comment by mfaizanse** [comment]
#154 same issue

**Comment by VTommyV** [comment]
Our AKS cluster in WestEurope also died on the 27th.  We can see from the activity graphs that all CPU activity died from that time.  We had no option but to recreate the AKS cluster :(

**Comment by mboersma** [comment]
The AKS preview service recently had an incident in west europe, and in recovering the region we ran into a problem where some clusters lost their state. The state itself has been backed up and can be recovered. 

If you are still affected by this, please contact Azure support and we will perform a remediation to recover your cluster data from backups.

**Comment by RobinHoody** [comment]
I've just scaled my cluster in Western Europe and had the same thing today. Strange. No big deal as it's only a dev cluster for testing in the preview. 

Are other regions experiencing the same issues? If not, i'll rebuild in another region.

**Comment by alexandervantrijffel** [comment]
We had the same issue 2 weeks ago and we notices that the routes to the node VMs were suddenly deleted rendering the nodes unavailable for Kubernetes.

**Comment by svenmueller** [comment]
Some problem for for me. The created cluster shows 3 nodes in the Azure UI, but kubectl shows "No resources" when doing `kubectl get nodes`.

- Terraform Azure provider 1.1.2
- Kubernetes `1.8.7`
- location `eastus`

```
resource "azurerm_kubernetes_cluster" "test_cluster" {
  name                   = "testcluster"
  location               = "East US"
  resource_group_name    = "testgroup"
  kubernetes_version     = "1.8.7"
  dns_prefix             = "aks"

  linux_profile {
    admin_username = "admin_1"

    ssh_key {
      key_data = "ssh-rsa ..."
    }
  }

  agent_pool_profile {
    name            = "default"
    count           = 1
    vm_size         = "Standard_A0"
    os_type         = "Linux"
  }

  service_principal {
    client_id     = "XXX"
    client_secret = "XXX"
  }

  tags {
    Environment = "playground"
  }
}
```

**Comment by svenmueller** [comment]
Same problem when using location `Central US`

**Comment by mboersma** [comment]
> Terraform Azure provider 1.1.2

This repository is just for support of the AKS preview service.

**Comment by svenmueller** [comment]
sure, but the AKS service is exactly the service which is failing here. i only mentioned the tool i used for creating the managed cluster

**Comment by mboersma** [comment]
@svenmueller ah, ok--just want to rule out TerraForm's `azurerm` provider as a source of the problem. Is this consistently reproducible, and does it happen if provisioning with just the `az` CLI? Could you open a ticket with Azure support so we can investigate that particular cluster / subscription?

**Comment by seanknox** [comment]
Closing due inactivity. Feel free to re-open if still an issue.

**Comment by ilanpillemer** [comment]
I still have the issues. cluster has 5, aks shows 5. kubetctl get nodes only gives 4.

**Comment by ilanpillemer** [comment]
How do I re-open since this is not fixed by any stretch of the imagination?

**Comment by seanknox** [comment]
@ilanpillemer please file a new issue.

**Comment by DOliana** [comment]
Since this is the only issue regarding the topic I wanted to share my solution. The issue occurred for a freshly created cluster with 40 nodes. What I did to fix was:

1. scale the cluster to one node
1. shutdown that node via Azure portal
1. start that node via Azure portal (observe, that kubectl get nodes now shows that node)
1. scale up to 40 nodes

Now everything works fine again. This occurred today both in westeurope and northeurope

**Comment by ilanpillemer** [comment]
just started happening to me again. 😢
